components/Chips/tests/unit/ChipsTests.m

```diff
@@ -394,4 +411,38 @@ - (void)testChipsWithDeleteEnabled {
   XCTAssertEqual(controlViewCount, (NSUInteger)1);
 }
 
+- (void)testChipViewDynamicTypeBehavior {
+  // Given
+  FakeMDCChipView *chipView = [[FakeMDCChipView alloc] init];
+  UIFont *titleFont = [UIFont systemFontOfSize:14.0 weight:UIFontWeightMedium];
+  MDCFontScaler *fontScaler = [[MDCFontScaler alloc] initForMaterialTextStyle:MDCTextStyleBody2];
+  titleFont = [fontScaler scaledFontWithFont:titleFont];
+  titleFont = [titleFont mdc_scaledFontAtDefaultSize];
+  chipView.titleFont = titleFont;
+  chipView.mdc_adjustsFontForContentSizeCategory = YES;
+  chipView.mdc_legacyFontScaling = NO;
+  chipView.titleLabel.text = @"Chip";
+  [NSNotificationCenter.defaultCenter
+      postNotificationName:UIContentSizeCategoryDidChangeNotification
+                    object:nil];
+  CGFloat defaultFontSize = chipView.titleLabel.font.pointSize;
+
+  // When
+  if (@available(iOS 10.0, *)) {
+    UIContentSizeCategory size = UIContentSizeCategoryExtraExtraLarge;
+    UITraitCollection *traitCollection =
+        [UITraitCollection traitCollectionWithPreferredContentSizeCategory:size];
+    chipView.traitCollectionOverride = traitCollection;
+  }
+  [NSNotificationCenter.defaultCenter
+      postNotificationName:UIContentSizeCategoryDidChangeNotification
+                    object:nil];
+  CGFloat fontSizeForExtraExtraLargeSizeCategory = chipView.titleLabel.font.pointSize;
+
+  // Then
+  if (@available(iOS 10.0, *)) {
+    XCTAssertGreaterThan(fontSizeForExtraExtraLargeSizeCategory, defaultFontSize);
```

Comment:
I think we can remove this `@available` check, it should be two CGFloats right?

Reply:
It only works for iOS 10 and above because we don't inject UIApplication for pre iOS 10.

Reply:
Both values are CGFloat, I see why we need to wrap the XCTAssertGreaterThan in the check?

Reply:
Please put the pre-iOS 10 behavior in the `else` block of the conditional.  If we expect it to always return the value for `.Large`  (or the identical value), that should be in the test.

Reply:
Done in ac3adc529191900c893a7edce591529bd2c0fddf.

Reply:
Follow up on this one, the whole test is wrapped for iOS 10.0+ in  865300029f6556aaf570620546e5bcb94e64d6f6 based on code review feedbacks.